RESTful.txt

```diff
@@ -0,0 +1,16 @@
+##Ice Cream psuedocode for RESTful conventions/best practices##
+
+* GET all flavors "/flavors"
+* GET specific flavor by id "/flavors/:id"
+* GET all ice cream types "/ictypes"
+* GET specific ice cream type by id "/ictypes/:id"
+* GET all flavors by ice cream type "/flavors/:id/ictypes"
```

Comment:
This would get all ice cream types matching a specific flavor (ex. all types that match the Chocolate or Strawberry flavor). If you reverse it "/ictypes/:id/flavors," you would get all flavors corresponding to an ice cream type. (ex. all flavors that match the containsNuts or glutenFree type.)